python/lsst/ip/isr/overscan.py

```diff
@@ -0,0 +1,504 @@
+            "LEG": "Fit Legendre polynomial to the longest axis of the overscan region",
+            "NATURAL_SPLINE": "Fit natural spline to the longest axis of the overscan region",
+            "CUBIC_SPLINE": "Fit cubic spline to the longest axis of the overscan region",
+            "AKIMA_SPLINE": "Fit Akima spline to the longest axis of the overscan region",
+            "MEAN": "Correct using the mean of the overscan region",
+            "MEANCLIP": "Correct using a clipped mean of the overscan region",
+            "MEDIAN": "Correct using the median of the overscan region",
+            "MEDIAN_PER_ROW": "Correct using the median per row of the overscan region",
+        },
+    )
+    order = pexConfig.Field(
+        dtype=int,
+        doc=("Order of polynomial or to fit if overscan fit type is a polynomial, " +
+             "or number of spline knots if overscan fit type is a spline."),
+        default=1,
+    )
+    numSigmaClip = pexConfig.Field(
+        dtype=float,
+        doc="Rejection threshold (sigma) for collapsing overscan before fit",
+        default=3.0,
+    )
+    overscanIsInt = pexConfig.Field(
+        dtype=bool,
+        doc="Treat overscan as an integer image for purposes of overscan.FitType=MEDIAN" +
+            " and overscan.FitType=MEDIAN_PER_ROW.",
+        default=True,
+    )
+
+
+class OverscanCorrectionTask(pipeBase.Task):
+    """Correction task for overscan.
+
+    This class contains a number of utilities that are easier to
+    understand and use when they are not embedded in nested if/else
+    loops.
+
+    Parameters
+    ----------
+    statControl : `lsst.afw.math.StatisticsControl`, optional
+        Statistics control object.
+    """
+    ConfigName = OverscanCorrectionTaskConfig
+    _DefaultName = "overscan"
+
+    def __init__(self, statControl=None, **kwargs):
+        super().__init__(**kwargs)
+        if statControl:
+            self.statControl = statControl
+        else:
+            self.statControl = afwMath.StatisticsControl()
+
+    def run(self, ampImage, overscanImage):
+        self.statControl.setNumSigmaClip(self.config.numSigmaClip)
+
+        if self.config.fitType in ('MEAN', 'MEANCLIP', 'MEDIAN'):
+            overscanResult = self.measureConstantOverscan(overscanImage)
+        elif self.config.fitType in ('MEDIAN_PER_ROW', 'POLY', 'CHEB', 'LEG',
+                                     'NATURAL_SPLINE', 'CUBIC_SPLINE', 'AKIMA_SPLINE'):
+            overscanResult = self.measureVectorOverscan(overscanImage, self.config.order)
+        else:
+            raise pexExcept.Exception('%s : %s an invalid overscan type' %
+                                      ("overscanCorrection", self.config.fitType))
+        overscanValue = overscanResult.overscanValue
+        maskArray = overscanResult.maskArray
+        isTransposed = overscanResult.isTransposed
+
+        if isinstance(overscanValue, float):
```

Comment:
This belongs in the `measureConstantOverscan` method, or in the above block that calls it. Similarly with the `else` block below.

Try to keep the `run` method as simple as possible.